src/jitlog/generator.lua

```diff
@@ -0,0 +1,358 @@
+local format = string.format
+
+local builtin_types = {
+  char = {size = 1, signed = true, printf = "%i", c = "char",  argtype = "char"},
+  
+  i8  = {size = 1, signed = true,  printf = "%i",   c = "int8_t",   argtype = "int32_t"},
+  u8  = {size = 1, signed = false, printf = "%i",   c = "uint8_t",  argtype = "uint32_t"},
+  i16 = {size = 2, signed = true,  printf = "%i",   c = "int16_t",  argtype = "int32_t"},
+  u16 = {size = 2, signed = false, printf = "%i",   c = "uint16_t", argtype = "uint32_t"},
+  i32 = {size = 4, signed = true,  printf = "%i",   c = "int32_t",  argtype = "int32_t"},
+  u32 = {size = 4, signed = false, printf = "%u",   c = "uint32_t", argtype = "uint32_t"},
+  i64 = {size = 8, signed = true,  printf = "%lli", c = "int64_t",  argtype = "int64_t"},
+  u64 = {size = 8, signed = false, printf = "%llu", c = "uint64_t", argtype = "uint64_t"},
+  
+  MSize  = {size = 4, signed = false,  printf = "%u", c = "uint32_t", argtype = "MSize"},
+  GCSize = {size = 4, signed = false,  printf = "%u", c = "GCSize", argtype = "GCSize"},
+  
+  int8_t  = "i8",  uint8_t  = "u8",
+  int16_t = "i16", uint16_t = "u16",
+  int32_t = "i32", uint32_t = "u32",
+  int64_t = "i64", uint64_t = "u64",
+
+  bitfield = "u32",
+  bool = "uint32_t",
+  
+  timestamp  = {size = 8, signed = false,  printf = "%llu", c = "uint64_t", writer = "timestamp_highres", noarg = true},
+  smallticks = {size = 4, signed = false,  printf = "%u",   c = "uint32_t", argtype = "uint64_t"},
+
+  GCRef      = {size = 4, signed = false, c = "GCRef", printf = "0x%llx", writer = "setref", ref = "gcptr32", ref64 = "gcptr64", argtype = "GCRef"},
+  --GCRef field with the value passed in as a pointer
+  GCRefPtr   = {size = 4, signed = false, c = "GCRef", printf = "0x%llx", writer = "setref", ref = "gcptr32", ref64 = "gcptr64", ptrarg = true, argtype = "void *"},
+  MRef       = {size = 4, signed = false, c = "MRef", printf = "0x%llx",  writer = "setref", ref = "ptr32", ref64 = "ptr64", ptrarg = true, argtype = "void *"},
+  -- Always gets widen to 64 bit since this is assumed not to be a gc pointer
+  ptr        = {size = 8, signed = false, c = "uint64_t", printf = "0x%llx", writer = "widenptr", ptrarg = true, argtype = "void *"},
+
+  string     = {vsize = true, printf = "%s", string = true,     c = "const char*", argtype = "const char *",  element_type = "char", element_size = 1},
+  stringlist = {vsize = true, printf = "%s", stringlist = true, c = "const char*", argtype = "const char *",  element_type = "char", element_size = 1},
+}
+
+if GC64 then
```

Comment:
This variable isn't defined is this file. Is that deliberate?

Reply:
Variables are global in Lua by default in this case its kind of being used like an environmental variable.

Reply:
Is it standard Lua practise to make use of this icky language feature? Can we not import the `GC64` variable to make clear to readers of this file where it came from?

Reply:
There isn't really such a thing as importing a variable in Lua